packages/neon_framework/lib/src/utils/request_manager.dart

```diff
@@ -134,6 +135,87 @@ class RequestManager {
         disableTimeout: disableTimeout,
       );
 
+  /// Executes a HTTP request for binary content.
+  Future<void> wrapBinary({
+    required Account account,
+    required String cacheKey,
+    required GetCacheParameters getCacheParameters,
+    required DynamiteRawResponse<Uint8List, dynamic> rawResponse,
+    required UnwrapCallback<Uint8List, Uint8List>? unwrap,
+    required BehaviorSubject<Result<Uint8List>> subject,
+    bool disableTimeout = false,
+  }) async =>
+      wrap<Uint8List, DynamiteRawResponse<Uint8List, dynamic>>(
+        account: account,
+        cacheKey: cacheKey,
+        subject: subject,
+        request: () async {
+          await rawResponse.future;
+          return rawResponse;
+        },
+        unwrap: (rawResponse) {
+          var data = rawResponse.response.body;
+          if (unwrap != null) {
+            data = unwrap(data);
+          }
+
+          return data;
+        },
+        serialize: (rawResponse) => base64.encode(rawResponse.response.body),
```

Comment:
As already mentioned there is way to much parsing going on.
Let's just save the binary data directly. IIRC the SQLite data type would be `BLOB`.

We would loose some performance as we would now need to utf8 en/decode normal api/webdav requests but they usually have a smaller payload anyways thus not being that bad.